android/src/main/java/com/swmansion/reanimated/layoutReanimation/SharedTransitionManager.java

```diff
@@ -442,7 +442,8 @@ protected void finishSharedAnimation(int tag) {
           if (key.equals(Snapshot.TRANSFORM_MATRIX)) {
             preparedValues.put(key, value);
           } else {
-            preparedValues.put(key, (double) PixelUtil.toDIPFromPixel((int) value));
+            float pixelsValue = value instanceof Integer ? (int) value : (Float) value;
```

Comment:
1. Here it's `(Float) value` but in line 274 it was `(float) value`, which one is correct?
2. Is it guaranteed that it's always either int or float?
3. Should we make a utility function out of this?

Reply:
You are right 👍 I made a util function from it. Resolved in https://github.com/software-mansion/react-native-reanimated/pull/4103/commits/5195fffb379a707cca35203988aec94b42bed76d